web-lib-funcs.pl

```diff
@@ -1968,29 +1968,113 @@ sub has_command
 return $rv;
 }
 
-=head2 make_date(seconds, [date-only], [fmt])
+=head2 make_date(seconds, [date-only-or-opts], [fmt])
 
 Converts a Unix date/time in seconds to a human-readable form, by default
 formatted like dd/mmm/yyyy hh:mm:ss. Parameters are :
 
 =item seconds - Unix time is seconds to convert.
 
-=item date-only - If set to 1, exclude the time from the returned string.
+=item date-only-or-opts - If set to 1, exclude the time from the returned string.
+
+In case this param is a hash reference use it for options in a new DateTime::Locale
+code or preserve the original, old logic
 
 =item fmt - Optional, one of dd/mon/yyyy, dd/mm/yyyy, mm/dd/yyyy or yyyy/mm/dd
 
 =cut
 sub make_date
 {
-&load_theme_library();
-if (defined(&theme_make_date) &&
-    !$main::theme_prevent_make_date && 
-    (($main::header_content_type eq "text/html" &&
-    $main::webmin_script_type eq "web") || 
-    $main::theme_allow_make_date)) {
-	return &theme_make_date(@_);
-	}
 my ($secs, $only, $fmt) = @_;
+eval "use DateTime::Locale";
+if (!$@) {
+	eval "use DateTime::TimeZone";
```

Comment:
Can you combine this and the next eval into a single statement? It would avoid the need to have double-nested if statements.

Reply:
Yes, all these were addressed.